Fails to compile on e.g. Arduino Mega 2560 due to INT5 macro

This bigger chip has more interrupts, so it defines [an `INT5` macro](https://github.com/vancegroup-mirrors/avr-libc/blob/06cc6ff5e6120b36f1b246871728addee58d3f87/avr-libc/include/avr/iomxx0_1.h#L321), which conflicts with Types.h:

```
.../MsgPack/MsgPack/Types.h:163:9: error: expected identifier before numeric constant
         INT5 = 0xE0, // same as NEGATIVE_FIXINT
```

This is really one of the reasons macros, especially such short ones, suck, but well, here we are. I can see two ways to fix this:
 - `#undef INT5` in `Types.h`. This is easy, but probably breaks some code out there that needs this particular bit *and* includes `MsgPack.h`
 - Rename the constant to something else that doesn't conflict. I *think* this does not change any public API, so this would probably be the preferred option.


To reproduce, compile the following sketch for the Arduino Mega 2560:

```
#include <MsgPack.h>
void setup() { }
void loop() { }
```

--- a/MsgPack/Types.h
+++ b/MsgPack/Types.h
@@ -9,6 +9,20 @@
     #include <string>
 #endif
 
+
+#ifdef INT5
+    // avoid conflict with interrrupt macro on e.g. Arduino Mega. This
+    // should not break code that rely on this constant, by keeping the
+    // value available as a C++-level constant. This first allocates a
+    // temporary constant, to capture the value before undeffing, and
+    // then defines the actual INT5 constant. Finally, INT5 is redefined
+    // as itself to make sure #ifdef INT5 still works.
+    static constexpr uint8_t INT5_TEMP_VALUE = INT5;
+    #undef INT5
+    static constexpr uint8_t INT5 = INT5_TEMP_VALUE;
+    #define INT5 INT5
+#endif
+
 #ifdef HT_SERIAL_MSGPACK_DISABLE_STL
     #include "util/ArxContainer/ArxContainer.h"
     #ifdef HT_SERIAL_MSGPACK_DISABLE_STL

